src/java.desktop/windows/native/libawt/windows/awt_PrintJob.cpp

```diff
@@ -690,6 +691,7 @@ Java_sun_awt_windows_WPageDialogPeer__1show(JNIEnv *env, jobject peer)
             }
             ::GlobalUnlock(setup.hDevMode);
         }
+        doIt = JNI_TRUE;
```

Comment:
Another option would be to return `JNI_TRUE` here and to return `JNI_FALSE` in the end of the function.

Reply:
Yes, it could have been done that way and my intiial fix in JBS is that only but I wanted to reinstate the flag as it is before [JDK-8307160](https://bugs.openjdk.org/browse/JDK-8307160) which was working till now..

Reply:
The only problem with the flag is that the data flow is not as clear.

I'm not insisting, it worked before and it will work in the future.

Reply:
The way it was "before" is that we always returned the value of "doIt". Why not restore that for consistency ?

Reply:
I believe that's what this PR is doing, it returns value of "doIt" at end, isn't it?